Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/util/ConnectionHelper.java

```diff
@@ -98,44 +92,22 @@ private boolean checkLocalHcid(String EntityHcid) {
         return formatHcid(EntityHcid).equals(formatHcid(getLocalHcid()));
     }
 
-    public String getHcidFromKeyedReference(List<KeyedReference> keyedRef) {
-        String homeCommunityId = null;
-        for (KeyedReference ref : keyedRef) {
-            if (ref.getTModelKey().equals(UDDIConstants.UDDI_HOME_COMMUNITY_ID_KEY)) {
-                homeCommunityId = ref.getKeyValue();
-                break;
-            }
-        }
-        return homeCommunityId;
-    }
-
-    private String getEntityName(BusinessEntity businessEntity) {
+    private String getOrganizationName(OrganizationType organization) {
         String homeCommunityName = null;
-        if (businessEntity.getName() != null && !businessEntity.getName().isEmpty()) {
-            homeCommunityName = businessEntity.getName().get(0).getValue();
+        if (organization != null && StringUtils.isNotBlank(organization.getName())) {
+            homeCommunityName =organization.getName();
         }
         return homeCommunityName;
     }
 
-    private List<KeyedReference> getKeyedReference(BusinessEntity entity) {
-        List<KeyedReference> keyedReference = new ArrayList<>();
-        if (entity != null && entity.getIdentifierBag() != null && entity.getIdentifierBag().getKeyedReference() != null
-            && !entity.getIdentifierBag().getKeyedReference().isEmpty()) {
-            keyedReference = entity.getIdentifierBag().getKeyedReference();
-        }
-        return keyedReference;
-    }
-
-    protected List<BusinessEntity> getAllBusinessEntities() {
-        List<BusinessEntity> businessEntities = null;
+    protected List<OrganizationType> getAllOrganizations() {
+        List<OrganizationType> organizations = null;
         try {
-            businessEntities = ConnectionManagerCache.getInstance().getAllBusinessEntities();
-        } catch (ConnectionManagerException ex) {
-            LOG.error(
-                "Exception while retrieving BusinessEntities from UDDIConnectionInfo: " + ex.getLocalizedMessage(),
-                ex);
+            organizations = WebServiceProxyHelper.getAllOrganizationsByMerge();
```

Comment:
We don't need to merge internalExchangeInfo and exchangeInfo. Remove the merge from WebserviceProxy, ExchangeManagerHelper and all related code.